tofu/tests/tests01_geom/test_04_sampling.py

```diff
@@ -248,24 +253,39 @@ def test25_sa_integ_map(ves_poly=VPoly, debug=1):
     assert (sz_p, npts_disc) == np.shape(sang)
     assert npts_disc > npts
 
-    sz_z = np.abs(limits_z[1] - limits_z[0]) // reso_z
-    sz_r = np.abs(limits_r[1] - limits_r[0]) // reso_r
+    sz_z = int(round(np.abs(limits_z[1] - limits_z[0]) / reso_z))
+    sz_r = int(round(np.abs(limits_r[1] - limits_r[0]) / reso_r))
     assert npts_sa == sz_z * sz_r
 
     sa_map_py = np.zeros((npts_sa, sz_p))
 
     for ii in range(npts_disc):
-        i_r = int((pts_disc[0, ii] - limits_r[0]) // reso_r)
-        i_z = int((pts_disc[1, ii] - limits_z[0]) // reso_z)
+        i_r = int(round((pts_disc[0, ii] - limits_r[0]) // reso_r))
+        i_z = int(round((pts_disc[1, ii] - limits_z[0]) // reso_z))
         ind_pol = int(i_r * sz_z + i_z)
-        if ind_pol == 39999:
-            print("solid angle = ", sang[0, ii], reso_phi[i_r])
         for pp in range(sz_p):
-            sa_map_py[ind_pol, pp] += sang[pp, ii] * np.pi * reso_phi[i_r]
+            sa_map_py[ind_pol, pp] += sang[pp, ii] * reso_phi[i_r]
+
+    if debug > 0:
+        fig = plt.figure(figsize=(14, 8))
+        ax = plt.subplot(121)
+        # import pdb; pdb.set_trace()
+        ax.scatter(pts[0, :], pts[1, :],
+                   marker="s", edgecolors="None",
+                   s=10, c=sa_map.flatten(),
+                   vmin=0, vmax=max(sa_map.max(), sa_map_py.max()))
+        ax.set_title("cython function")
+        ax = plt.subplot(122)
+        ax.scatter(pts[0, :], pts[1, :],
+                   marker="s", edgecolors="None",
+                   s=10, c=sa_map_py.flatten(),
+                   vmin=0, vmax=max(sa_map.max(), sa_map_py.max()))
+        ax.set_title("python reconstruction")
+        plt.savefig("comparaison")
 
     print("reconstructed sa map =", sa_map_py)
     print("cython computed = ", sa_map)
-    assert np.allclose(sa_map_py, sa_map)
+    assert np.allclose(sa_map_py, sa_map, atol=1e-05)
```

Comment:
1e-5 is not a very accurate computation.

Is it because the solid angle formula is different between cython and python ?

Reply:
no, but i did try to go higher in order and got a better approximation (didn't verify the order of approx). Let me check

Reply:
Question:
* why are we comparing an aproximate formula to an exact formula ?
* * Why not compare approximate vs approximate and exact vs exact ?